### PR TITLE
Adds GPG signing and verification

### PR DESCRIPTION
**Pull request by pittma**
Addresses #89 

Hey there!

Here's my work so far w.r.t. implementing signing and verification. As it stands now, signing is available for both requests and acceptances, and we can verify those at `pull` time. I wanted to share now to see if there were any major contentions with the path I've gone down before I went through the rote repetition of adding signing functionality to the other comment-based commands.

Cheers!

### CODE REVIEW

**Comment by pittma** [comment]
CLA signed.

**Comment by ojarjur** [comment]
@pittma Thanks for taking this on!

It will take me some time to go through the full change, but I plan to give you feedback by the end of the week.

File: review/gpg/signable.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+// Verify verifies the signatures on the request and its comments with the
+// given key.
+func Verify(key string, s Signable) error {
+	// Retrieve the pointer.
+	sigPtr := s.Signature()
+	// Copy its contents.
+	sig := *sigPtr
+	// Overwrite the value with the placeholder.
+	*sigPtr = placeholder
```
Comment (ojarjur) [inline review/gpg/signable.go]: I think maybe we should add a defer statement here to set the value of `*sigPtr` back to `sig` prior to the method returning.

File: repository/repo.go [hunk 2]
```diff
@@ -185,4 +189,18 @@ type Repo interface {
 	// we merely ensure that their history graph includes every commit that we
 	// intend to keep.
 	PullNotesAndArchive(remote, notesRefPattern, archiveRefPattern string) error
+
+	// MergeNotesAndArchive merges the notes and archives from `remote` into
+	// the local branch.
+	MergeNotesAndArchive(remote, notesRefPattern,
```
Comment (ojarjur) [inline repository/repo.go]: I wonder if we should split this in to two separate methods; `MergeNotes` and `MergeArchives`. The reason is that we will soon want to add something like `MergeForks`, and maybe even `MergeKeys`, and we don't want to have to combine all of those together into a single method name.

File: review/gpg/signable.go (hunk 1 above)
Comment (pittma) [reply review/gpg/signable.go]: Whew. Seriously good point :smile:.

File: review/review.go [hunk 3]
```diff
@@ -646,7 +691,7 @@ func (r *Review) AddComment(c comment.Comment) error {
 // review will be added to the 'refs/devtools/archives/reviews' ref prior
 // to being rewritten. That ensures the review history is kept from being
 // garbage collected.
-func (r *Review) Rebase(archivePrevious bool) error {
+func (r *Review) Rebase(archivePrevious, sign bool) error {
```
Comment (ojarjur) [inline review/review.go]: Can you please add a RebaseAndSign method here similarly to how you added for the Repo interface?

That will both prevent backwards-incompatible changes, and also keep the Review API consistent with the Repo API.

The current method body could just be a helper that both `Rebase` and `RebaseAndSign` invoke.

Comment (pittma) [reply review/review.go]: On it!

**Pull request merged**